### DISCUSSION TITLE
Auto rotate security tokens & start prompt in new chats once one fills up

### DISCUSSION THREAD

**Discussion by PlutoniaX** [acheong08/ChatGPT#460] [Q&A] [answered]
Hi @acheong08 firstly a huge thank you for this phenomenal project it's really helped me automate a lot of daily processes with GPT. 

I have been using to summarise books and would like to:
1) auto-rotate the session security token across several ChatGPT accounts so that i can workaround the hourly limit, and 
2) auto-start a New Chat as an existing one fills up which leads to Internal Errors or cannot generate response error

Your (and anyone's) guidance would be much appreciated!

**Comment by acheong08** [comment, marked as answer]
This would break conversations which is the main feature of ChatGPT. I have added some code which allows specifying the sesstion_token when calling ask(). You can cycle sessions in your implementation but that probably won't be part of this library.

**Reply by PlutoniaX** [reply]
ah thank you! will use ask() in my code. On 2, what i meant is that currently if i pipe my texts to the same chat, it fills up the chat session after a while and i'd have to manually go into chatGPT and delete the chat on the left hand panel. So i was wondering if there is a way to initiate a new chat ID / session after say every 20 chunks of text, or if there is a better way to do this?

**Reply by acheong08** [reply]
You can call the `reset_chat()` function to make a new chat ID

**Reply by PlutoniaX** [reply]
ah got it. thank you so much!

**Reply by PlutoniaX** [reply]
@acheong08 is this the right way to swap the session token? it's giving me error "tls_client.exceptions.TLSClientExeption: net/http: request canceled (Client.Timeout or context cancellation while reading body)"
i also used ask() and similar errors occur. Feels like i am missing something.
```
1stToken = "xxxxxxx"
2ndToken = "xxxxxxx"
tempToken = ''
count=0

chatbot = Chatbot({
      "session_token": 1stToken
    }, conversation_id=None, parent_id=None)
    
for text in texts:
  if (count==40):
          chatbot.refresh_session(session_token=2ndToken)
          # swap the 1st & 2nd tokens
          tempToken=2ndToken
          2ndToken=1stToken
          1stToken=tempToken
          count=0

   response = chatbot.ask("prompt. \""+text+"\"", conversation_id=None, parent_id=None)
   summary = response['message']

   count+=1
```
